src/tests/plugins/sendmail/test_rules.py

```diff
@@ -0,0 +1,250 @@
+            datetime.datetime(2021, 5, 21, 10, tzinfo=utc)
+        ),
+
+        # Tests for timezone quirks
+        (  # Test sending on leap day
+            datetime.datetime(2020, 2, 27, 9, tzinfo=utc),
+            None,
+            'UTC',
+            Rule(date_is_absolute=False, offset_is_after=True, send_offset_days=2, send_offset_time=datetime.time(hour=9)),
+            datetime.datetime(2020, 2, 29, 9, tzinfo=utc)
+        ),
+        (  # Test timezone far off from UTC
+            NZ.localize(datetime.datetime(2021, 5, 17, 22)),
+            None,
+            'NZ',
+            Rule(date_is_absolute=False, offset_is_after=True, send_offset_days=1, send_offset_time=datetime.time(hour=9)),
+            NZ.localize(datetime.datetime(2021, 5, 18, 9))
+        ),
+        (  # Test across DST change
+            Berlin.localize(datetime.datetime(2021, 10, 29, 16, 30)),
+            None,
+            'Europe/Berlin',
+            Rule(date_is_absolute=False, offset_is_after=True, send_offset_days=4, send_offset_time=datetime.time(hour=2, minute=30)),
+            Berlin.localize(datetime.datetime(2021, 11, 2, 2, 30))
+        ),
+        (  # Test ambiguous time at DST change
+            Berlin.localize(datetime.datetime(2021, 10, 29, 18, 30)),
+            None,
+            'Europe/Berlin',
+            Rule(date_is_absolute=False, offset_is_after=True, send_offset_days=2, send_offset_time=datetime.time(hour=2, minute=30)),
+            datetime.datetime(2021, 10, 31, 1, 30, tzinfo=utc)
+        ),
+        (  # Test non-existing time at DST change
+            Berlin.localize(datetime.datetime(2021, 3, 29, 14, 30)),
+            None,
+            'Europe/Berlin',
+            Rule(date_is_absolute=False, offset_is_after=False, send_offset_days=1, send_offset_time=datetime.time(hour=2, minute=30)),
+            datetime.datetime(2021, 3, 28, 1, 30, tzinfo=utc)
+        ),
+
+    ])
+def test_sendmail_rule_send_time(event_from, event_to, event_tz, rule, expected, event):
+    if event_from:
+        event.date_from = event_from
+        event.save()
+
+    if event_to:
+        event.date_to = event_to
+        event.save()
+
+    event.settings.timezone = event_tz
+
+    rule.event = event
+    rule.save()
+    m = ScheduledMail.objects.filter(rule=rule).get()
+
+    assert m.computed_datetime.astimezone(event.timezone) == expected
+
+
+@pytest.mark.django_db
+@scopes_disabled()
+def test_sendmail_rule_recompute(event):
+    event.has_subevents = True
+    event.save()
+    se1 = event.subevents.create(name="meow", date_from=dt_now)
+
+    rule = event.rule_set.create(date_is_absolute=False, offset_is_after=False, send_offset_days=2, send_offset_time=datetime.time(4, 30))
+
+    se1.date_from += datetime.timedelta(days=2)
+    se1.save()
+
+    expected = dt_now.replace(hour=4, minute=30, second=0, microsecond=0)
+
+    m = ScheduledMail.objects.filter(rule=rule).first()
+    m.recompute()
```

Comment:
instead of recomputing manually, maybe run the periodic task to check that it does so automatically?